packages/shared/src/components/filters/TagsFilter.tsx

```diff
@@ -58,10 +62,29 @@ export default function TagsFilter({
   );
   const { data: searchResults } = useQuery<SearchTagsData>(
     searchKey,
-    () => request(`${apiUrl}/graphql`, SEARCH_TAGS_QUERY, { query }),
-    {
-      enabled: query?.length > 0,
+    async () => {
+      const data = await request<SearchTagsData>(
+        `${apiUrl}/graphql`,
+        SEARCH_TAGS_QUERY,
+        { query },
+      );
+
+      if (!query) {
+        return data;
+      }
+
+      trackEvent({
+        target_type: TargetType.SearchTag,
+        target_id: TargetId.TagsFilter,
```

Comment:
target_id and target_type are not required here. I left a comment for Sab on the taxonomy